### DISCUSSION TITLE
Append content before body closing

### DISCUSSION THREAD

**Discussion by barsdeveloper** [joomla/joomla-cms#34880] [Q&A] [answered]
I am inside a module template (/templates/mytemplate/html/mod_login/modal.php) and I want to split the module in 2 parts: one that is output directly (and appears in its module position as defined by the template) and one that is written at the end of body (or at the beginning, doesn't matter, it just matters to be outside any html element because I have issues with the bootstrap modal z-index).

```PHP
<button type="button" class="btn" data-bs-toggle="modal"
    Some content
</button>
<?php
$app->appendBody(
    <<<HTML
<div id="login-form-<?php echo $module->id; ?>-wrapper"
	class="modal fade" tabindex="-1">
    Some content
</div>
HTML
);
```

This doesn't work, I think body is reset somewhere else. What is the correct way to achieve this? Without using JS to move the element client side.

**Comment by dgrammatiko** [comment, marked as answer]
The code already exists: https://github.com/joomla/joomla-cms/pull/32062

Would be helpful for all the modals or other non-essential HTML to be thrown at the bottom...

**Reply by barsdeveloper** [reply]
That's exactly the issue and not really an edge case not to mention it would not be the only use case. Maybe closing it wasn't the best decision. Also oldest pull request is from 2016 two weeks is not so much. Thanks anyway.

**Reply by dgrammatiko** [reply]
> Maybe closing it wasn't the best decision.

I think I closed it as the release leader wasn't convinced on both the use cases and also the implementation. You're more than welcome to resurrect that PR
